src/Configuration/Dotenv/ParsedVariable.hs

```diff
@@ -0,0 +1,55 @@
+module Configuration.Dotenv.ParsedVariable (ParsedVariable(..),
+                                            VName,
+                                            VValue(..),
+                                            VContents,
+                                            VFragment(..),
+                                            interpolateParsedVariables) where
+
+import Control.Monad (foldM)
+#if MIN_VERSION_optparse_applicative(0,13,0)
+import Data.Functor ((<$>))
+#endif
+import Control.Applicative ((<|>))
+import System.Environment (lookupEnv)
+
+data ParsedVariable
+  = ParsedVariable VName VValue deriving (Show, Eq)
+
+type VName = String
+
+data VValue
+  = Unquoted VContents
+  | SingleQuoted VContents
+  | DoubleQuoted VContents deriving (Show, Eq)
+
+type VContents = [VFragment]
+
+data VFragment
+  = VInterpolation String
+  | VLiteral String deriving (Show, Eq)
+
+interpolateParsedVariables :: [ParsedVariable] -> IO [(String, String)]
```

Comment:
OK, I've looked at the code and it seems that:

1. The whole `Configuration.Dotenv.ParsedVariable` module is an internal helper, so it's sort of OK to have `IO` there.

2. The only place this `interpolateParsedVariables` function is used is the `parseFile`, which is already in `MonadIO`.

So as long as `interpolateParsedVariables` only lookups things and does not modify environment, I'm OK with the current signatures. We of course could grab all the environment as a `Map String String` and then have a reader monad instead of `IO`, but for this use case, it looks like an unjustified and slower solution.